### PR TITLE
Migrate to Gradle

### LINKED ISSUE #173: Migrate to use Gradle build
https://github.com/jenkinsci/gradle-jpi-plugin
- allows usage of groovy to make plugin code less verbose
- no changes for end consumers
- better testability using groovy meta classes etc

### PR DESCRIPTION
**Pull request by kageiit**
Fixes #173

Interop with maven:
`mvn hpi` - `./gradlew jpi`
`mvn install` - `./gradlew publishToMavenLocal`
`mvn publish` - `./gradlew publish`
`mvn hpi:run` - `./gradlew server`

More details here:
https://github.com/jenkinsci/gradle-jpi-plugin
https://wiki.jenkins-ci.org/display/JENKINS/Gradle+JPI+Plugin

### CODE REVIEW

**Comment by sectioneight** [comment]
Can you update the readme to mention the new dev flow?

**Comment by kageiit** [comment]
> Can you update the readme to mention the new dev flow?

Sure. Waiting for travis to sanity check it. Will update the README

File: .gitignore [hunk 1]
```diff
@@ -3,3 +3,8 @@ target/
 .idea
 .DS_Store
 *.iml
+
+.gradle
+local.properties
+.DS_Store
```
Comment (sectioneight) [inline .gitignore]: This is already on line 4 :P

Comment (kageiit) [inline .gitignore]: Fixed

**Pull request merged**